### PR TITLE
added new API proposal [DO NOT MERGE]

### PR DESCRIPTION
**Pull request by gfrogat**
!DO NOT MERGE!

We only use the review functionality for discussing the API for now.

### CODE REVIEW

File: malevo/swagger_v2.yml [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+tags:
+- name: "run"
+  description: ""
+- name: "view"
+  description: ""
+schemes:
+- "https"
+- "http"
+paths:
+  /run:
```
Comment (thinkh) [inline malevo/swagger_v2.yml]: Why not call it `runs` since it returns all available runs?

File: malevo/swagger_v2.yml [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    type: "object"
+    properties:
+      runId:
+        type: "integer"
+        format: "int64"
+      dataset:
+        $ref: "#/definitions/Dataset"
+      hyperparam:
+        $ref: "#/definitions/HyperParam"
+      epochdata:
```
Comment (thinkh) [inline malevo/swagger_v2.yml]: This must be an array (currently it is a single epoch only).

Swagger output:
```
     "epochdata": {
        "epochId": 0,
        "fold": "string",
        "confmat": [
          0
        ]
      }
```

File: malevo/swagger_v2.yml [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+      responses:
+        200:
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/Run"
+        400:
+          description: "Invalid ID supplied"
+        404:
+          description: "Run not found"
+  /view:
```
Comment (thinkh) [inline malevo/swagger_v2.yml]: Why not call it `views` since it returns all available views?

File: malevo/swagger_v2.yml (hunk 2 above)
Comment (gfrogat) [reply malevo/swagger_v2.yml]: yes, you're right. I'll fix it.

File: malevo/swagger_v2.yml (hunk 1 above)
Comment (gfrogat) [reply malevo/swagger_v2.yml]: missed that, sorry

File: malevo/swagger_v2.yml (hunk 3 above)
Comment (gfrogat) [reply malevo/swagger_v2.yml]: missed that, sorry

**Comment by gfrogat** [comment]
> I think we should separate the epoch data from the run meta data. Currently /run would return all runs with all epochs. So I suggest the following (missing) routes:

> /run/<id>/epochdata -> returns all epoch data for the run

I would have simply returned the runs with empty epochdata arrays when we call `/runs` (also doesn't violate the schema definition), but it's probably better to have a separate call.

> PUT /view with multiple run ids to create a new view -> return new view id
I didn't add a way to set the views yet because I'm still thinking about the automatic logging.

**Comment by thinkh** [comment]
> I would have simply returned the runs with empty epochdata arrays when we call /runs (also doesn't violate the schema definition), but it's probably better to have a separate call.

An empty array for the epoch data is also fine for me. 

> I didn't add a way to set the views yet because I'm still thinking about the automatic logging.

Ok, take your time. 😄

**Comment by thinkh** [comment]
Discussion moved to https://github.com/gfrogat/confusionflow/issues/1.

**Pull request closed without merging**